message.go

```diff
@@ -238,9 +239,15 @@ func (m *Message) Unpack(src []byte) error {
 // locked by the caller.
 func (m *Message) wrapErrorUnpack(src []byte) error {
 	if fieldID, err := m.unpack(src); err != nil {
-		return &UnpackError{
+		subfields := []string{}
+		var unpackErr *iso8583errors.UnpackError
+		if errors.As(err, &unpackErr) {
+			subfields = unpackErr.FieldIDs
+		}
+		return &iso8583errors.UnpackError{
 			Err:        err,
 			FieldID:    fieldID,
+			FieldIDs:   append([]string{fieldID}, subfields...),
 			RawMessage: src,
 		}
```

Comment:
```suggestion
		return &iso8583errors.UnpackError{
			Err:        err,
			FieldID:    fieldID,
			RawMessage: src,
		}
```

please, read the main message that will explain the suggestion